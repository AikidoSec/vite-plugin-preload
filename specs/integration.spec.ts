import fs from "fs";

describe("vite - build", () => {
  test("html contain preload links", async () => {
    const htmlFile = "examples/react-demo/dist/index.html";

    expect(fs.existsSync(htmlFile));
    const htmlContent = fs.readFileSync(htmlFile, {
      encoding: "utf8",
      flag: "r",
    });

    expect(htmlContent).toMatchSnapshot("index_with_preloads");
  });
});
