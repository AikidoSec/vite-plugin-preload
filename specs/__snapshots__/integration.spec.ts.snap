// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite - build html contain preload links: index_with_preloads 1`] = `
"<html>
  <head>
    <title>React example</title>
    <script
      type=\\"module\\"
      crossorigin=\\"\\"
      src=\\"/assets/index.06e372d5.js\\"
    ></script>
    <link rel=\\"modulepreload\\" href=\\"/assets/vendor.4fa92e17.js\\" />
    <link rel=\\"stylesheet\\" href=\\"/assets/index.5de8cc00.css\\" />
    <link rel=\\"modulepreload\\" href=\\"/assets/index.470dc361.js\\" />
    <link rel=\\"modulepreload\\" href=\\"/assets/index.79b31f97.js\\" />
    <link rel=\\"modulepreload\\" href=\\"/assets/index.f2bd501d.js\\" />
    <link rel=\\"stylesheet\\" href=\\"/assets/index.2921fa01.css\\" />
    <link rel=\\"stylesheet\\" href=\\"/assets/index.e3157e37.css\\" />
    <link rel=\\"stylesheet\\" href=\\"/assets/index.ef9b644c.css\\" />
  </head>
  <body>
    <div id=\\"root\\"></div>
  </body>
</html>
"
`;
